libqtile/backend/wayland/drawer.py

```diff
@@ -55,13 +61,23 @@ def draw(
         if offsetx > self._win.width:  # type: ignore
             return
 
-        # Only attempt to run RecordingSurface's operations if it actually has
-        # some. self.surface.ink_extents() computes the bounds of the current
-        # list of operations. If any of the bounds are not 0.0 then the surface
-        # has operations and we should paint them.
-        if not any(not math.isclose(0.0, i) for i in self.surface.ink_extents()):
+        # We can't test for the surface's ink_extents here on its own as a completely
+        # transparent background would not show any extents but we may still need to
+        # redraw (e.g. if a Spacer widget has changed position and/or size)
+
+        # Check if current rect has changed since previous draw
+        current_rec = (offsetx, offsety, width, height)
+        rect_changed = any([p != c for p, c in zip(self.previous_rect, current_rec)])
```

Comment:
You can just do `rect_changed = self.previous_rect == current_rec`

Reply:
what the hell was I thinking when I wrote that line?